tun_server.go

```diff
@@ -0,0 +1,111 @@
+package node
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+
+	"github.com/AutoRoute/tuntap"
+
+	"github.com/AutoRoute/node/types"
+)
+
+var requestHeader = "hello"
+var beginIPBlock = net.ParseIP("6666::0")
+var endIPBlock = net.ParseIP("6666::6666")
+
+type TunServer struct {
+	data        DataConnection
+	tun         TCPTun
+	amt         int64
+	nodes       map[string]types.NodeAddress
+	connections map[types.NodeAddress]bool
+	currIP      int
+	err         chan error
+}
+
+func NewTunServer(d DataConnection, amt int64) *TunServer {
+	i, err := tuntap.Open("tun%d", tuntap.DevTun)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return &TunServer{d, i, amt, make(map[string]types.NodeAddress), make(map[types.NodeAddress]bool), 0, make(chan error, 1)}
+}
+
+func (ts *TunServer) Error() chan error {
+	return ts.err
+}
+
+func (ts *TunServer) connect(connectingNode string) {
+	ip := fmt.Sprintf("6666::%d", ts.currIP)
```

Comment:
You should probably pass a subnet as a flag and allocate ip addresses out of that (since they should really match the ip's given to this server or a private range).